### PR TITLE
update default values for RocksDB config

### PR DESCRIPTION
**Pull request by tabokie**
### What is changed, added or deleted? (Required)

https://github.com/tikv/tikv/pull/15055
https://github.com/tikv/tikv/pull/15115
https://github.com/tikv/tikv/pull/15174

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by Oreoxmt** [comment]
@tabokie Are the compatibility changes involved in this PR introduced in v7.3?

**Comment by tabokie** [comment]
7.3 branch is not cut, all the changes on master branch before the cut will be in 7.3 release.

File: tikv-configuration-file.md [hunk 1]
```diff
@@ -516,7 +516,7 @@ Configuration items related to the sharing of block cache among multiple RocksDB
 ### `capacity`
 
 + The size of the shared block cache.
-+ Default value: 45% of the size of total system memory
++ Default value: 45% of the size of total system memory when `storage.engine="raft-kv"`, 30% of the size of total system memory when `storage.engine="partitioned-raft-kv"`
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:519]:
```suggestion
+ Default value:

    + When `storage.engine="raft-kv"`, the default value is 45% of the size of total system memory.
    + When `storage.engine="partitioned-raft-kv"`, the default value is 30% of the size of total system memory.

```

File: tikv-configuration-file.md [hunk 2]
```diff
@@ -1153,12 +1153,12 @@ Configuration items related to RocksDB
 ### `max-total-wal-size`
 
 + The maximum RocksDB WAL size in total, which is the size of `*.log` files in the `data-dir`.
-+ Default value: `"4GB"`
++ Default value: `"4GB"` when `storage.engine="raft-kv"`, `1` when `storage.engine="partitioned-raft-kv"`
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1156]:
```suggestion
+ Default value:

    + When `storage.engine="raft-kv"`, the default value is `"4GB"`.
    + When `storage.engine="partitioned-raft-kv"`, the default value is `1`, which means that the WAL is disabled.
```

File: tikv-configuration-file.md [hunk 3]
```diff
@@ -1153,12 +1153,12 @@ Configuration items related to RocksDB
...
 
 ### `stats-dump-period`
 
 + The interval at which statistics are output to the log.
-+ Default value: `10m`
++ Default value: `"10m"` when `storage.engine="raft-kv"`, `"0"` when `storage.engine="partitioned-raft-kv"`
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1161]:
```suggestion
+ Default value:

    + When `storage.engine="raft-kv"`, the default value is `"10m"`.
    + When `storage.engine="partitioned-raft-kv"`, the default value is `"0"`.
```

File: tikv-configuration-file.md [hunk 4]
```diff
@@ -1268,8 +1268,8 @@ Configuration items related to RocksDB
 >
 > This feature is experimental. It is not recommended that you use it in the production environment. This feature might be changed or removed without prior notice. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.
 
-+ Specifies the total memory limit of `memtable` for all RocksDB instances in a single TiKV. The default value is 25% of the memory of the machine. It is recommended to configure a memory of at least 5 GiB. This configuration only takes effect for Partitioned Raft KV (`storage.engine`=`"partitioned-raft-kv"`).
-+ Default value: 25%
++ Specifies the total memory limit of `memtable` for all RocksDB instances in a single TiKV. Zero stands for no limit.
++ Default value: Zero when `storage.engine="raft-kv"`, 20% of the size of total system memory when `storage.engine="partitioned-raft-kv"`
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1272]:
```suggestion
+ Specifies the total memory limit of `memtable` for all RocksDB instances in a single TiKV. `0` means no limit.
+ Default value:

    + When `storage.engine="raft-kv"`, the default value is `0`, which means no limit.
    + When `storage.engine="partitioned-raft-kv"`, the default value is 20% of the size of total system memory.

```

File: tikv-configuration-file.md [hunk 5]
```diff
@@ -1537,7 +1537,7 @@ Configuration items related to `rocksdb.defaultcf`, `rocksdb.writecf`, and `rock
     - `3`: Can be read by TiKV v2.1 and later versions. Changes the encoding of the keys in index blocks.
     - `4`: Can be read by TiKV v3.0 and later versions. Changes the encoding of the values in index blocks.
     - `5`: Can be read by TiKV v6.1 and later versions. Full and partitioned filters use a faster and more accurate Bloom filter implementation with a different schema.
-+ Default value: `2`
++ Default value: `2` when `storage.engine="raft-kv"`, `5` when `storage.engine="partitioned-raft-kv"`
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1540]:
```suggestion
+ Default value:

    + When `storage.engine="raft-kv"`, the default value is `2`.
    + When `storage.engine="partitioned-raft-kv"`, the default value is `5`.
```

File: tikv-configuration-file.md (hunk 2 above)
Comment (Oreoxmt) [reply tikv-configuration-file.md:1156]: https://github.com/pingcap/docs-cn/pull/14598/files#r1275870175

**Comment by Oreoxmt** [comment]
@tabokie 
Could you please invite a tech reviewer? Thanks!

**Review by tonyxuqqi — APPROVED** [review]

**Review by Oreoxmt — APPROVED** [review]

File: tikv-configuration-file.md [hunk 6]
```diff
@@ -1153,12 +1157,18 @@ Configuration items related to RocksDB
 ### `max-total-wal-size`
```
Comment (hfxsd) [inline tikv-configuration-file.md:1157]: 中文版这部分没有更新

**Review by hfxsd — APPROVED** [review]

**Pull request merged**